### ISSUE TITLE
My opinion on point number 7

### ISSUE THREAD

**Issue by portasynthinca3** [Widdershin/programmers-oath#159]
It states:
> I will remember that I do not write code for computers, but for people.

As I understand it, this point calls for readable code over code that performs a bit better but looks way worse. I mostly agree with that, but sometimes when working with extremely limited platforms like microcontrollers it can be hard to follow this rule while keeping reasonable performance.

  1. Is my interpretation of the rule correct?
  2. If so, can it be okay to break it in certain cases?
  3. If so, is it worth it to reflect this possibility in the rule?

**Comment by Widdershin** [comment]
Hi @portasynthinca3.

Thanks for your interest in this project.

There was quite an in-depth discussion around this tenet in #22, and I wrote quite a bit there around my intent when I wrote this tenet.

> As I understand it, this point calls for readable code over code that performs a bit better but looks way worse. I mostly agree with that, but sometimes when working with extremely limited platforms like microcontrollers it can be hard to follow this rule while keeping reasonable performance.

The essence of this tenet is an appeal to remember the human impact of our work, rather than focusing on abstract technical ideals.

I think in this scenario, it would be reasonable at times to sacrifice the readability of code for an optimization that had sufficient benefit to people who interact with the system (developers, users, etc).

In my interpretation, this tenet would also encourage adding a comment to any code like this, to help future developers understand the optimized code.

This tenet would discourage optimization for the sake of optimization, especially if the optimizations have no real-world positive impact on anyone but they make the code harder to work with.

I think in hindsight, this tenet was poorly worded, and can clearly be quite confusing. It might be worth removing it outright, since there's quite a lot of overlap with the meaning of other tenets.
